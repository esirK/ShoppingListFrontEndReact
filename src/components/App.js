import React, { Component } from 'react';
import RegisterForm from '../containers/register_form'

export default class App extends Component {
  render() {
    return (
      <div className="container">
        <RegisterForm />
      </div>
    );
  }
}