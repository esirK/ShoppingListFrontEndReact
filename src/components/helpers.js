import React from 'react';
import {FlatButton} from 'material-ui';
import TextField from 'material-ui/TextField';


export function validate(values){
	const errors = {};
	if (!values.username){
		errors.username = 'Username Required';
	}
	if (!values.email){
		errors.email = 'Email Required';
	}
	if (!values.password){
		errors.password = 'Password Required';
	}
	if (values.password && values.password.length < 6){
		errors.password = 'Passwords must be at least 6 Characters Long';
	}
	if (!values.confirm_password){
		errors.confirm_password = 'Enter confirmation Password';
	}
	if(values.password !== values.confirm_password){
		errors.confirm_password = 'Passwords do not match';
	}
	return errors;
}

export function itemActions(context){
	let actions = [
		<FlatButton
		  label="Cancel"
		  primary={true}
		  onClick={context.handleClose}
		/>,
		<FlatButton
		  label="Submit"
		  primary={true}
		  onClick={context.handleSubmit}
		  disabled={context.props.isUpdating||context.props.isCreatingNewShoppingList||context.props.isShareling}
		/>,
	];
	return actions;
}

export function newShoppinglistChildren(context){
	let children = [
		<TextField
		  hintText="Shoppinglist Name"
		  errorText={context.props.error}
		  value={context.state.name}
		  fullWidth={true}
		  key='name'
		  onChange={context.handleTitleChange}
		/>,
		<TextField
		  hintText="Description of the shopping list"
		  multiLine={true}
		  rows={2}
		  rowsMax={4}
		  fullWidth={true}
		  value={context.state.description}
		  key='desc'
		  onChange={context.handleDescChange}
		/>
	];
	return children;
}
export function updateChildren(context){
	let children = [
		<TextField
		  hintText="New Item Name"
		  errorText={context.props.error}
		  value={context.state.name}
		  fullWidth={true}
		  onChange={context.handleNameChange}
		  key='name'
		/>,
		<TextField
		  hintText="New Price of the Item"
		  multiLine={true}
		  fullWidth={true}
		  errorText={context.state.priceError}
		  value={context.state.price}
		  onChange={context.handlePriceChange}
		  key='price'
		/>,
		<TextField
		  hintText="Quantity of the Item"
		  multiLine={true}
		  fullWidth={true}
		  errorText={context.state.quantityError}
		  value={context.state.quantity}
		  onChange={context.handleQuantityChange}
		  key='quantity'
		/>,
		<p key='error' className='alert alert-danger'>{context.state.error}</p>
	];
	return children;
}
export function shareShoppinglistChildren(context){
	let children = [
		<TextField
		  hintText="Email address to share with"
		  errorText={context.props.error}
		  value={context.state.email}
		  fullWidth={true}
		  onChange={context.handleEmailChange}
		  key='email'
		/>,
	];
	return children;
}
