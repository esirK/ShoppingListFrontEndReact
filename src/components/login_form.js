import React, {Component} from 'react';


export default class LoginForm extends Component{
    render(){
        return(
            <div>
                <p>I am Mr Meesix Look at me</p>
            </div>
        );
    }
}