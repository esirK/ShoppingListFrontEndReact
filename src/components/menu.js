import React, {Component} from 'react';
import {connect} from 'react-redux';

import MenuItem from 'material-ui/MenuItem';
import TextField from 'material-ui/TextField';
import {FlatButton} from 'material-ui';

import Dialog from 'material-ui/Dialog';
import Snackbar from 'material-ui/Snackbar';

import IconMenu from 'material-ui/IconMenu';
import IconButton from 'material-ui/IconButton';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

import {Logout, update_profile, hideSnackBar} from '../actions';

export class Menu extends Component{
	constructor(props){
		super(props);
		this.state ={
			conf_logout: false,
			open_profile: false,
			username: '',
			password: '',
			old_password: '',
			openSb: false
		};
		this.logout = this.logout.bind(this);
		this.profile = this.profile.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleLogout = this.handleLogout.bind(this);
		this.handlePasswordChange = this.handlePasswordChange.bind(this);
		this.handleUsernameChange = this.handleUsernameChange.bind(this);
		this.handleProfileUpdate = this.handleProfileUpdate.bind(this);
	}
	logout(){
		this.setState({conf_logout: true});
	}
	profile(){
		this.setState({open_profile: true});
		this.props.hideSnackBar();
	}
	handleLogout(){
		this.props.Logout();
		this.setState({conf_logout: false});
		//reload page on Logout
		window.location.reload();
	}
	handlePasswordChange(e){
		this.setState({password: e.target.value});
	}
	handleUsernameChange(e){
		this.setState({username: e.target.value});
	}
	handleProfileUpdate(){
		this.props.update_profile({username:this.state.username, password:this.state.password});
		this.setState({open_profile: false});
		this.setState({'username':''});
		this.setState({'password':''});
		this.setState({'old_password':''});
	}
	handleClose(){
		this.setState({conf_logout: false});
		this.setState({open_profile: false});
		this.setState({'username':''});
		this.setState({'password':''});
		this.setState({'old_password':''});
		this.props.hideSnackBar();
	}
	render(){
		let actions = [
			<FlatButton
				label="Cancel"
				primary={true}
				onClick={this.handleClose}
			/>,
			<FlatButton
				label="OK"
				primary={true}
				onClick={this.handleLogout}
			/>,
		];
		let profile_actions = [
			<FlatButton
				label="Cancel"
				primary={true}
				onClick={this.handleClose}
			/>,
			<FlatButton
				label="Update"
				primary={true}
				onClick={this.handleProfileUpdate}
			/>,
		];
		const profile_children = [
			<TextField
			  hintText="New Username"
			  value={this.state.username}
			  fullWidth={true}
			  key='name'
			  onChange={this.handleUsernameChange}
			/>,
			<TextField
			  hintText="Old Password"
			  multiLine={true}
			  fullWidth={true}
			  value={this.state.password}
			  key='old_p'
			  onChange={this.handlePasswordChange}
			  type={'password'}
			/>,
			<TextField
			  hintText="New Password"
			  multiLine={true}
			  fullWidth={true}
			  key='new_p'
			  onChange={this.handlePasswordChange}
			  type="password"
			>
			</TextField>
		];
		return(
			<div>
				<IconMenu
					iconButtonElement={
						<IconButton><MoreVertIcon /></IconButton>
					}
					targetOrigin={{horizontal: 'left', vertical: 'top'}}
					anchorOrigin={{horizontal: 'right', vertical: 'top'}}
				>
					<MenuItem primaryText="Profile" onClick={this.profile} />
					<MenuItem primaryText="Sign out" onClick={this.logout}/>
				</IconMenu>
				<Dialog
					actions={actions}
					modal={false}
					open={this.state.conf_logout}
					onRequestClose={this.handleClose}
				>
         			Confirm Logout?
				</Dialog>
				<Dialog open={this.state.open_profile}
					title="Update Personal details"
					actions={profile_actions}
					children={profile_children}
					onRequestClose={this.handleClose}
				>
				</Dialog>
				<Snackbar
					open={this.props.openSb}
					message={this.props.message}
					autoHideDuration={3000}
					onRequestClose={this.handleRequestClose}
				/>
			</div>
		);
	}
}
function mapStateToProps(state) {
	return{
		message: state.authentication.message,
		openSb: state.authentication.openSb,
	};
}
export default connect(mapStateToProps, {Logout, update_profile, hideSnackBar})(Menu);