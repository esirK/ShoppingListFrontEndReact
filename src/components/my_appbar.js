import React, { Component } from 'react';

import {connect} from 'react-redux';
import AppBar from 'material-ui/AppBar';

import Menu from './menu';

export class MyAppBar extends Component{
	render(){
		return(
			<AppBar title="Shopping List App" showMenuIconButton={false}
				iconElementRight={this.props.isAuthenticated?<Menu />:null}
			/>
		);
	};
}
function mapStateToProps(state, ownProps) {
	return{
		isAuthenticated: state.authentication.isAuthenticated,
	};
}
export default connect(mapStateToProps, null)( MyAppBar);