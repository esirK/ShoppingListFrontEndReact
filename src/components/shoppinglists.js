import React, { Component } from 'react';
import {Card, CardActions, CardHeader, CardText, FlatButton} from 'material-ui';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';
import {connect} from 'react-redux';

import Dialog from 'material-ui/Dialog';
import TextField from 'material-ui/TextField';
import Snackbar from 'material-ui/Snackbar';



import {addNewShoppingList, getShoppingLists, deleteShoppingList, updateShoppingList, resetErrors} from '../actions';

class ShoppingLists extends Component{
	constructor(props){
		super(props);
		console.log('starting again');
		this.state = {
			addOpen: false,
			name: '',
			description: '',
			error: props.error,
			open: false,
			message: '',
			openUpdate: false
		};
		this.handleFabClick = this.handleFabClick.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleTitleChange = this.handleTitleChange.bind(this);
		this.handleDescChange = this.handleDescChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
		this.handleRequestClose = this.handleRequestClose.bind(this);
		this.handleUpdate = this.handleUpdate.bind(this);
		this.handleUpdateShoppingList = this.handleUpdateShoppingList.bind(this);
	}
	componentDidMount(){
		console.log('state ',this.state);
		this.props.getShoppingLists();
	}
	handleFabClick(){
		console.log('Clicked Fab', this.state.addOpen);
		this.setState({ addOpen: true });
	}
	handleClose(){
		this.setState({ name: '' });
		this.setState({ description: ''});
		this.setState({ addOpen: false });
		this.setState({openUpdate: false});
		this.props.resetErrors();
	}
	handleSubmit(){
		this.props.addNewShoppingList({
			'name': this.state.name,
			'description': this.state.description
		});
		this.setState({ open: true});
		this.setState({ name: '' });
		this.setState({ description: ''});
	}

	handleTitleChange(e){
		this.setState({ name: e.target.value });
	}
	handleDescChange(e){
		this.setState({ description: e.target.value });
	}
	deleteList(id){
		console.log('deleting...', id);
		this.props.deleteShoppingList(id, this.setState({open: true}));
	}
	handleUpdateShoppingList(id){
		this.setState({ openUpdate: true, id});
	}
	handleUpdate(){
		this.props.updateShoppingList(this.state.id, {
			'new_name': this.state.name,
			'description': this.state.description
		}, this.setState({open: true}));
		this.setState({ name: '' });
		this.setState({ description: ''});
		// this.setState({ openUpdate: false});
	}
	handleRequestClose(){
		this.setState({open: false});
	}
	render(){
		let cards = [];
		const actions = [
			<FlatButton
			  label="Cancel"
			  primary={true}
			  onClick={this.handleClose}
			/>,
			<FlatButton
			  label="Submit"
			  primary={true}
			  onClick={this.handleSubmit}
			/>,
		  ];	
		  const children = [
			  <TextField
      			hintText="Shoppinglist Name"
				errorText={this.props.error}
				value={this.state.name}
				fullWidth={true}
				key='name'
				onChange={this.handleTitleChange}
			/>,
			<TextField
				hintText="Description of the shopping list"
				multiLine={true}
				rows={2}
				rowsMax={4}
				fullWidth={true}
				value={this.state.description}
				key='desc'
				onChange={this.handleDescChange}
			/>
		  ];
		  const update_actions = [
			<FlatButton
			  label="Cancel"
			  primary={true}
			  onClick={this.handleClose}
			/>,
			<FlatButton
			  label="Update"
			  primary={true}
			  onClick={this.handleUpdate}
			/>,
		  ];	
		  const update_children = [
			  <TextField
      			hintText="New Shoppinglist Name"
				errorText={this.props.error}
				value={this.state.name}
				fullWidth={true}
				key='name'
				onChange={this.handleTitleChange}
			/>,
			<TextField
				hintText="New Description of the shopping list"
				multiLine={true}
				rows={2}
				rowsMax={4}
				fullWidth={true}
				value={this.state.description}
				key='desc'
				onChange={this.handleDescChange}
			/>
		  ];
		if(this.props.error && !this.state.addOpen){
			return(
				<div id="cards">
					<p>Got a {this.props.error} while Loading Your Shoppinglists</p>
				</div>
			);
		}
		if(this.props.isLoading){
			return(
				<div id="cards">
					<p>Loading...</p>
				</div>
			);
		}
		else{
			this.props.shoppinglists.map((shoppinglist)=>
				cards.push(
					<Card key={shoppinglist.id}  onExpandChange={this.handleExpandChange}>
						<CardHeader
							title={shoppinglist.name}
							subtitle={shoppinglist.items.length+' Items Available'}
							actAsExpander={true}
							showExpandableButton={true}
						/>
						<CardText expandable={true}>
        				{shoppinglist.description}
						</CardText>
						<CardActions>
							<FlatButton label="View" primary={true}/>
							<FlatButton label="Update" onClick={()=>this.handleUpdateShoppingList(shoppinglist.id)}/>
							<FlatButton name='delete' label="Delete" secondary={true} onClick={()=> this.deleteList(shoppinglist.id)}/>
						</CardActions>
					</Card>
				)
			);
			return(
				<div id="cards">
					<Dialog open={this.state.addOpen}
						title="Add Shoppinglist"
						actions={actions}
						children={children}
					>
					</Dialog>
					<Dialog open={this.state.openUpdate}
						title="Update Shoppinglist"
						actions={update_actions}
						children={update_children}
					>
					</Dialog>
			    {cards}
					<div id="fab">
						<FloatingActionButton secondary={true} onClick={this.handleFabClick}>
							<ContentAdd />
						</FloatingActionButton>
					</div>
					<Snackbar
						open={this.state.open}
						message={this.props.message||this.props.error }
						autoHideDuration={4000}
						onRequestClose={this.handleRequestClose}
					/>
				</div>
			);
		}
	}
}
function mapStateToProps(state){
	return {
		isLoading: state.shoppinglists.isLoading,
		shoppinglists: state.shoppinglists.shoppinglists,
		error: state.shoppinglists.error,
		message: state.shoppinglists.message,
	};
}
export default connect(mapStateToProps, {addNewShoppingList, getShoppingLists, deleteShoppingList, updateShoppingList, resetErrors}) (ShoppingLists);