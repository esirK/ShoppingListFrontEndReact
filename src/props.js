export const shoppinglist_props = {
	activateFab:jest.fn(),
	addFab:false,
	isLoading: false,
	shoppinglists: [],
	all_shoppinglists: [],
	error: '',
	message: '',
	openSb: false,
	addFab: false,
	openUpdate: false,
	resetErrors: jest.fn(),
	getShoppingLists: jest.fn(),
};
export const shoppinglist_item_props = {
	activateAddItem:jest.fn(),
	activateUpdateItem:jest.fn(),
	deleteShoppinglistItem:jest.fn(),
	hideSnackBar:jest.fn(),
	error:false,
	addFab:false,
	isOpenUpdateItem:false,
	match:{
		isExact: true,
		params:{id:166},
		path:'/:id/shoppinglist_items',
		url:'/166/shoppinglist_items',
	},
	data:{
		items:[]
	},
	message:'',
	openAddItem:'',
	openSb:false,
	isLoading: false,
	resetErrors: jest.fn(),
	viewShoppingList: jest.fn(),
};