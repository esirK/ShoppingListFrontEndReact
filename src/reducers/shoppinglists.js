import * as types from '../constants/actiontypes';

const initState = {
	isLoading: false,
	isCreatingNewShoppingList: false,
	shoppinglists: [],
	error: false,
	message: '',
	openSb: false
};

export default function(state = initState, action){
	switch(action.type){
	case types.GETTING_SHOPPINGLISTS_STARTED:
		console.log('getting shoppinglist started');
		return{
			...state, isLoading:true
		};
	case types.SHOPPINGLISTS_LOADED_SUCCESSFULLY:
		return{
			...state, shoppinglists:action.response,
			isLoading:false
		};
	case types.CREATING_SHOPPINGLISTS_STARTED:
	    console.log('Creation of a new shoppinglist started');
		return {
			...state, isCreatingNewShoppingList:true,
			open: true
		};
	case types.SHOPPINGLISTS_CREATED_SUCCESSFULY:
		return {
			...state, shoppinglists:action.response,
			isCreatingNewShoppingList:false, open: false
		};
	case types.SHOPPINGLISTS_DELETED_SUCCESSFULY:
		return {
			...state, message: action.message
		};
	case types.SHOPPINGLISTS_UPDATED_SUCCESSFULY:
		return {
			...state, shoppinglists:action.response,
			message: action.message
		};
	case types.ERROR_ENCOUNTERED:
		return{
			...state, isLoading:false, error: action.error
		};
	case types.CLEAR_ERRORS:
		console.log('Clearing errors', state);
		return{
			...state, error:false, message: '', openSb: false
		};

	default:
		return state;
	}
}