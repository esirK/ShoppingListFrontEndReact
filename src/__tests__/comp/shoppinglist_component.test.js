import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {ShoppingLists} from '../../components/shoppinglists';
import {shoppinglist_props} from '../../props';

describe('Shoppinglist component', ()=>{
	const myAppBar = shallow(<ShoppingLists {...shoppinglist_props}/>);
	it('renders Shoppinglist without crashing', ()=>{
		expect(shallowToJson(myAppBar)).toMatchSnapshot();
	});
});
