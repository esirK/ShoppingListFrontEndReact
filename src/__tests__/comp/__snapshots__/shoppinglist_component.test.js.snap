// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoppinglist component renders Shoppinglist without crashing 1`] = `
<div
  id="cards"
>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="Search"
    fullWidth={true}
    hintText="Shoppinglist Name"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
  />
  <br />
  <ul
    className="pagination pagination-lg pagination-centered"
  />
  <div
    id="fab"
  >
    <FloatingActionButton
      disabled={false}
      mini={false}
      onClick={[Function]}
      secondary={true}
      zDepth={2}
    >
      <ContentAdd />
    </FloatingActionButton>
  </div>
  <Snackbar
    autoHideDuration={3000}
    message=""
    onRequestClose={[Function]}
    open={false}
  />
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="OK"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
  >
    Confirm Delete?
  </Dialog>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Submit"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={true}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title="Share with"
  >
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      fullWidth={true}
      hintText="Email address to share with"
      key="email"
      multiLine={false}
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value=""
    />
  </Dialog>
</div>
`;
